src/components/googlemap/CustomMap.tsx

```diff
@@ -36,73 +27,86 @@ class CustomMap extends Component<IpropsCustomMap, IstateCustomMap> {
         },
         zoom: 6
       },
-      markers: [{ id: 1, lat: props.lat, lng: props.lng, text: props.text }]
+      markers: [{
+        id: 1,
+        lat: props.lat,
+        lng: props.lng,
+        text: props.text,
+      }]
     };
   }
 
   addMapMarker = (
     event: FormEvent<HTMLInputElement>,
-    params: { id: number; lat: number; lng: number; text: string }
+    params: Marker,
   ): void => {
+    const { markers } = this.state;
     event.preventDefault();
     // id from the cosmos :)
-    const id = this.state.markers.length + 1;
+    const id = markers.length + 1;
     this.setState({
       markers: [
-        ...this.state.markers,
-        { id: id, lat: params.lat, lng: params.lng, text: params.text }
-      ]
+        ...markers,
+        {
+          id: id,
+          lat: params.lat,
+          lng: params.lng,
+          text: params.text,
+        },
+      ],
     });
-    console.log("markers", this.state.markers);
+    console.log("markers", markers);
   };
 
   removeMapMarker = (id: number): void => {
-    const filter = this.state.markers.filter(marker => marker.id !== id);
+    const { markers } = this.state;
+    const filter = markers.filter((marker: Marker) => marker.id !== id);
 
     console.log("filter:", filter);
 
     this.setState({
       // ...this.state,
-      markers: this.state.markers.filter(marker => marker.id !== id)
+      markers: markers.filter((marker: Marker) => marker.id !== id)
     });
   };
 
   render() {
+    const { markers, settings } = this.state;
     const mapInlineStyles: CSSProperties = {
       width: "100%",
       height: "500px"
     };
 
     let markersContent;
 
-    if (this.state.markers) {
-      markersContent = this.state.markers.map(marker => {
-        return (
-          <MapMarker
-            key={marker.id}
-            id={marker.id}
-            lat={marker.lat}
-            lng={marker.lng}
-            text={marker.text}
-            onRemoveMarkerHandler={this.removeMapMarker}
-          />
-        );
-      });
+    if (markers) {
+      markersContent = markers.map((marker: Marker) => (
+        <MapMarker
+          key={marker.id}
+          id={marker.id}
+          lat={marker.lat}
+          lng={marker.lng}
+          text={marker.text}
+          onRemoveMarkerHandler={this.removeMapMarker}
+        />
+      ));
     } else {
       markersContent = "spinner";
     }
 
     return (
       <div>
-        <CustomMapForm addMarkerEventHandler={this.addMapMarker} />
+        <CustomMapForm
+          addMarkerEventHandler={this.addMapMarker}
+        />
         <h2 className="container">Localizations</h2>
         <div style={mapInlineStyles}>
           <GoogleMapReact
             bootstrapURLKeys={{
-              key: "AIzaSyAl6Fg23N7K6UdHwgWVVf8DCsT8FUhQyh4"
+                key: GOOGLE_MAP_KEY,
```

Comment:
Klucze do API nigdy nie powinny się znaleźć w repozytorium. Najlepsze miejsce dla kluczy to zmienne środowiskowe.